presto-main/src/main/java/com/facebook/presto/sql/InterpretedFunctionInvoker.java

```diff
@@ -146,4 +169,45 @@ private static RuntimeException propagate(Throwable throwable)
         throwIfUnchecked(throwable);
         throw new RuntimeException(throwable);
     }
+
+    public void writeOutput(Type type, BlockBuilder output, Object argument)
+    {
+        switch (type.getJavaType().getSimpleName()) {
```

Comment:
Instead of switching on the Java type, can you switch on instances of the built-in type?  e.g. take type.getBase() and check it against well known types in StandardTypes.java